### PR TITLE
copr: fix Max/Min bug when comparing signed and unsigned int64 (#10167)

### LINKED ISSUE #10158: copr bug: Agg func Max/Min got bug when compare signed and unsigned int64
### What version of TiKV are you using?

master

### What operating system and CPU are you using?

Ubuntu 18.04

model name      : AMD Ryzen 5 PRO 4650U with Radeon Graphics
### Steps to reproduce

  drop table t1;
  CREATE TABLE t1  (COL1 bigint(16) UNSIGNED NULL DEFAULT NULL);
  INSERT INTO t1 VALUES (11333449897206147134);
  INSERT INTO t1 VALUES (2420174916247255494);
  INSERT INTO t1 VALUES (3899490809029152765);
  select max(col1), sum(isnull(col1)), count(1) from t1;

### What did you expect?
expect: 11333449897206147134
### What did happened?
but got: 3899490809029152765

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #10167 to release-4.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tikv repo:
git pr https://github.com/tikv/tikv/pull/10616
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tikv.git pr/10616:release-4.0-bb064d642429
```

---

### What problem does this PR solve?

Issue Number: close #10158 

Problem Summary: The max function does not consider whether int64 is unsigned when comparing, and always assumes that the input is signed. Therefore, the comparison result of 11333449897206147134 and 3899490809029152765 is problematic.

    drop table t1;
    CREATE TABLE t1 (COL1 bigint(16) UNSIGNED NULL DEFAULT NULL);
    INSERT INTO t1 VALUES (11333449897206147134);
    INSERT INTO t1 VALUES (2420174916247255494);
    INSERT INTO t1 VALUES (3899490809029152765);
    select max(col1), sum(isnull(col1)), count(1) from t1;
    
    Use unistore(Following result is correct):
    
    tidb(localhost:4001) > select max(col1), sum(isnull(col1)), count(1) from t1;
    +----------------------+-------------------+----------+
    | max(col1)            | sum(isnull(col1)) | count(1) |
    +----------------------+-------------------+----------+
    | 11333449897206147134 |                 0 |        3 |
    +----------------------+-------------------+----------+
    1 row in set (0.01 sec)

    use tikv(Following result is wrong):
    
    tidb(localhost:4000) > select max(col1), sum(isnull(col1)), count(1) from t1;
    +---------------------+-------------------+----------+
    | max(col1)           | sum(isnull(col1)) | count(1) |
    +---------------------+-------------------+----------+
    | 3899490809029152765 |                 0 |        3 |
    +---------------------+-------------------+----------+
    1 row in set (0.00 sec)
### What is changed and how it works?

What's Changed: Add a new implementation for Int comparison. We use a const template argument to determinate whether input is unsigned or not.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Integration test: test_aggr_unsigned_signed_int

Side effects: no
```release-note
copr: fix Max/Min bug when comparing signed and unsigned int64
```

### CODE REVIEW

File: components/tidb_query/src/lib.rs [hunk 1]
```diff
@@ -26,6 +26,7 @@
 #![allow(clippy::no_effect)]
 #![feature(box_patterns)]
 #![feature(matches_macro)]
+#![feature(const_generics)]
```
Comment (guo-shaoge) [inline components/tidb_query/src/lib.rs:29]: Looks like the compiler version is old.

```
   Compiling tidb_query v0.0.1 (/home/guo-shaoge/go/src/github.com/pingcap/tikv/components/tidb_query)
warning: the feature `const_generics` is incomplete and may cause the compiler to crash
  --> components/tidb_query/src/lib.rs:29:12
   |
29 | #![feature(const_generics)]
   |            ^^^^^^^^^^^^^^
   |
   = note: `#[warn(incomplete_features)]` on by default
```

Not sure is it ok to use this feature. @skyzh

Comment (skyzh) [reply components/tidb_query/src/lib.rs]: Yes, we should find another way to implement this.

**Review by skyzh — APPROVED** [review]

**Review by wshwsh12 — APPROVED** [review]

**Pull request merged**